internal/services/cdn/cdn_frontdoor_origin_resource.go

```diff
@@ -320,21 +333,32 @@ func resourceCdnFrontDoorOriginUpdate(d *pluginsdk.ResourceData, meta interface{
 	if d.HasChange("private_link") {
 		// I need to get the profile SKU so I know if it is valid or not to define a private link as
 		// private links are only allowed in the premium sku...
-		profileId := parse.NewFrontDoorProfileID(id.SubscriptionId, id.ResourceGroup, id.ProfileName)
-		profile, err := profileClient.Get(ctx, profileId.ResourceGroup, profileId.ProfileName)
+		profileId := profiles.NewProfileID(id.SubscriptionId, id.ResourceGroup, id.ProfileName)
+
+		profileResp, err := profileClient.Get(ctx, profileId)
 		if err != nil {
-			if utils.ResponseWasNotFound(profile.Response) {
-				return fmt.Errorf("retrieving parent %s: not found", profileId)
+			if response.WasNotFound(profileResp.HttpResponse) {
+				d.SetId("")
+				return nil
 			}
+			return fmt.Errorf("retrieving %s: %+v", profileId, err)
+		}
+
+		profileModel := profileResp.Model
+
+		if profileModel == nil {
+			return fmt.Errorf("profileModel is 'nil'")
+		}
 
-			return fmt.Errorf("retrieving parent %s: %+v", profileId, err)
+		if profileModel.Properties == nil {
+			return fmt.Errorf("profileModel.Properties is 'nil'")
 		}
```

Comment:
do we care about this if we get the SKU? I can't see it used (at least from this diff)
```suggestion
```

Reply:
Yes, we need the `sku` to determine if it is valid set a `private_link` property or not, only premium sku's allow private links to be set.